addons/crm/models/res_partner.py

```diff
@@ -77,19 +77,19 @@ def _compute_meeting(self):
             meeting_data = self.env.cr.fetchall()
 
             # Create a dict {partner_id: event_ids} and fill with events linked to the partner
-            meetings = {p.id: set() for p in all_partners}
+            meetings = {p_id: set() for p_id in set(m[0] for m in meeting_data)}
             for m in meeting_data:
                 meetings[m[0]].add(m[1])
 
             # Add the events linked to the children of the partner
-            all_partners.read(['parent_id'])
-            for p in all_partners:
+            partners_with_meetings = self.browse(meetings.keys())
+            for p in partners_with_meetings:
```

Comment:
Could probably concatenate both lines ? No need to use an intermediary variable, is it ?